contracts/social-tips/src/contract.rs

```diff
@@ -0,0 +1,1375 @@
+    let response = querier.query_application_link_owners(
+        Some(application.clone()),
+        Some(handle.clone()),
+        Some(PageRequest {
+            key: None,
+            limit: Uint64::new(1),
+            offset: owner_index,
+            reverse: false,
+            count_total: false,
+        }),
+    )?;
+
+    let serialized_coins = serialize_coins(&funds);
+
+    return if !response.owners.is_empty() {
+        let owner = response.owners.first().unwrap().user.to_string();
+
+        Ok(Response::new()
+            .add_attribute(ATTRIBUTE_ACTION, ACTION_SEND_TIPS)
+            .add_attribute(ATTRIBUTE_TIP_COLLECTED, "false")
+            .add_attribute(ATTRIBUTE_TIP_CLAIMER, &owner)
+            .add_attribute(ATTRIBUTE_TIP_AMOUNT, serialized_coins)
+            .add_message(BankMsg::Send {
+                amount: funds,
+                to_address: owner,
+            }))
+    } else {
+        let config = CONFIG.load(deps.storage)?;
+        let tips = pending_tips();
+
+        let user_sent_pending_tips_count = tips
+            .idx
+            .sender
+            .prefix(sender.clone())
+            .range_raw(deps.storage, None, None, Order::Ascending)
+            .count();
+
+        // Ensure that the sender can not spam the chain with multiple tips to different users.
+        if user_sent_pending_tips_count >= config.max_sent_pending_tips as usize {
+            return Err(ContractError::ToManySentPendingTips {});
+        }
+
+        let user_pending_tips_count = tips
+            .prefix((application.clone(), handle.clone()))
+            .range_raw(deps.storage, None, None, Order::Ascending)
+            .count();
+
+        // Ensure that the user don't have to many pending tips.
+        if user_pending_tips_count >= config.max_pending_tips as usize {
+            return Err(ContractError::ToManyPendingTipsForUser {
+                application,
+                handle,
+            });
+        }
+
+        let key = (application.clone(), handle.clone(), sender.clone());
+        let replaced = tips.may_load(deps.storage, key.clone())?;
+
+        tips.replace(
+            deps.storage,
+            key,
+            Some(&PendingTip {
+                sender,
+                amount: funds,
+                block_height: env.block.height,
+            }),
+            replaced.as_ref(),
+        )?;
+
+        let mut response = Response::new()
+            .add_attribute(ATTRIBUTE_ACTION, ACTION_SEND_TIPS)
+            .add_attribute(ATTRIBUTE_TIP_COLLECTED, "true")
+            .add_attribute(ATTRIBUTE_TIP_AMOUNT, serialized_coins);
+
+        // Send back the funds of the replaced tip.
+        if let Some(replaced_tip) = replaced {
+            response = response.add_message(BankMsg::Send {
+                amount: replaced_tip.amount,
+                to_address: replaced_tip.sender.to_string(),
+            });
+        }
+
+        Ok(response)
+    };
```

Comment:
Since it is related to `x/bank`, I would have a integration test to test the final balance of the sender.

Reply:
There are already unit tests to ensure that the proper message has ben added to the response, I think is more than enough to ensure the correct behaviour of the contract.